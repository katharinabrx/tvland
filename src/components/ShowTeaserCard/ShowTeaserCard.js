import React from "react";
import { Link } from "react-router-dom";
import "./ShowTeaserCard.scss";

const ShowTeaserCard = ({ showId, showName, showImage }) => {
  return (
    <article className="ShowTeaserCard">
      <Link to={`/shows/${showId}`} key={showId} className="ShowTeaserCard__link">
        <img src={showImage} alt={showName} className="ShowTeaserCard__card" />
        <div className="ShowTeaserCard__title">{showName}</div>
      </Link>
    </article>
  );
};

export default ShowTeaserCard;
