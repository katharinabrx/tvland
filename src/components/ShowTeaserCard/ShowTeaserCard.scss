@import "../../styles/styles.scss";
.ShowTeaserCard {
  text-decoration: none;

  &__card {
    margin-bottom: 1rem;
    width: 100%;
    @extend .card;
    text-decoration: none;
  }
  &__link {
    text-decoration: none;
  }
  &__title {
    color: theme-color("primary", "dark");
    font-weight: 700;
    @include text-setting("md");
  }
}
