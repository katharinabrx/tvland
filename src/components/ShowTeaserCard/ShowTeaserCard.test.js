import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ShowTeaserCard from "./ShowTeaserCard";

it("expect to render ShowTeaserCard component with empty input", () => {
  expect(toJson(shallow(<ShowTeaserCard />))).toMatchSnapshot();
});

it("expect to render ShowTeaserCard component with placeholder image", () => {
  const mockShow = [
    {
      id: 1,
      name: "Show Name 1",
      image: undefined
    }
  ];
  expect(
    toJson(
      shallow(
        <ShowTeaserCard showId={mockShow.id} showName={mockShow.name} showImage={undefined} />
      )
    )
  ).toMatchSnapshot();
});

it("expect to render ShowTeaserCard component with image", () => {
  let image = [{ medium: "http://static.tvmaze.com/uploads/images/medium_portrait/1/4600.jpg" }];
  expect(
    toJson(shallow(<ShowTeaserCard showId="1" showName="Test Name" showImage={image} />))
  ).toMatchSnapshot();
});
