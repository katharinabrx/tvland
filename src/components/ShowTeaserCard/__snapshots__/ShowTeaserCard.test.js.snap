// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render ShowTeaserCard component with empty input 1`] = `
<article
  className="ShowTeaserCard"
>
  <Link
    className="ShowTeaserCard__link"
    to="/shows/undefined"
  >
    <img
      className="ShowTeaserCard__card"
    />
    <div
      className="ShowTeaserCard__title"
    />
  </Link>
</article>
`;

exports[`expect to render ShowTeaserCard component with image 1`] = `
<article
  className="ShowTeaserCard"
>
  <Link
    className="ShowTeaserCard__link"
    key="1"
    to="/shows/1"
  >
    <img
      alt="Test Name"
      className="ShowTeaserCard__card"
      src={
        Array [
          Object {
            "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/1/4600.jpg",
          },
        ]
      }
    />
    <div
      className="ShowTeaserCard__title"
    >
      Test Name
    </div>
  </Link>
</article>
`;

exports[`expect to render ShowTeaserCard component with placeholder image 1`] = `
<article
  className="ShowTeaserCard"
>
  <Link
    className="ShowTeaserCard__link"
    to="/shows/undefined"
  >
    <img
      className="ShowTeaserCard__card"
    />
    <div
      className="ShowTeaserCard__title"
    />
  </Link>
</article>
`;
